File: backend/src/main/java/com/example/pojo/Room.java
package com.example.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Room {
    @TableId(type = IdType.AUTO)
    private Integer id;
    private String url;
    private String title;
    private String city;
    private String district;
    private String road;
    private String community;
    private String uptime;
    private Date savetime;
    private Double price;
    private String rentalmethod;
    private String housetype;
    private String roomtype;
    private String areasize;
    private String decoratetype;
    private String ori;
    private String renttimeinfo;
    private String floorinfo;
    private Boolean hasrefrigerator;
    private Boolean haswashingmachine;
    private Boolean hasbalcony;
    private Boolean hascook;
    private Boolean hasairconditioner;
    private Boolean haswaterheater;
    private Boolean hasbed;
    private Boolean haswardrobe;
    private Boolean hassofa;
    private Boolean haswifi;
    private String bedroomnum;
    private String bathroomnum;
    private String hallsnum;
    private String pic;
    private String traffic;
    private String description;
    private int pageViews;
}

File: admin-v1/src/types/serviceEntity/Room.ts
interface Room {
  id: number;
  url: string;
  title: string;
  city: string;
  district: string;
  road: string;
  community: string;
  uptime: string;
  savetime: string;
  price: string;
  rentalmethod: string;
  housetype: string;
  roomtype: string;
  areasize: string;
  decoratetype: string;
  ori: string;
  renttimeinfo: string;
  floorinfo: string;
  hasrefrigerator: number;
  haswashingmachine: number;
  hasbalcony: number;
  hascook: number;
  hasairconditioner: number;
  haswaterheater: number;
  hasbed: number;
  haswardrobe: number;
  hassofa: number;
  haswifi: number;
  bedroomnum: string;
  bathroomnum: string;
  hallsnum: string;
  pic: string;
  traffic: string;
  description: string;
  pageViews: number;
  correctPic?:string[]
}

enum RoomBookingState{
  BOOKED = 1,
  FINISH 
}
const RoomBookingStateMap = {
  [RoomBookingState.BOOKED]:'已预约',
  [RoomBookingState.FINISH]:'已完成'
}

interface RoomBooking {
  id: number;
  userId: number;
  houseId: number;
  appointmentTime:string,
  createTime: string;
  visitTime: string;
  requirementDesc: string;
  history: string;
  state:RoomBookingState
}
interface RoomBookingOrder extends RoomBooking {
  room:Room
}
enum RentState {
    PAYING =1,
    PAID,
    CHECKIN,
    DONE
}
const RentStateMap = {
  [RentState.PAID]:{
    title:'待入住',
    subtitle:'待入住'
  },
  [RentState.CHECKIN]:{
    title:'已入住',
    subtitle:'已入款'
  },
  [RentState.DONE]:{
    title:'已完成',
    subtitle:'已完成'
  },
}
interface RentInfo {
  id: number;
  userId: number;
  houseId: number;
  state: RentState;
  createTime: string;
  payTime?: string;
  signTime?: string;
  checkinTime?: string;
  admin: string;
  payment: number;
  // 租期(月为单位)
  tenancy: number;
  history: string;
  serviceIds:string
}

interface RentOrderInfo extends RentInfo{
  room:Room
}

interface FilterOption {
  district?: string;
  traffic?: string;
  community?: string;
  road?:string,
  ori?: string;
  lowestPrice?:number,
  highestPrice?:number,
  lowestAreaSize?: number;
  highestAreaSize?: number;
  floorInfo?: string;
  orderBy?:  'price' | 'areaSize';
  sort?:'desc' | 'asc'
}

interface RoomVisitHistory{
  date:Date,
  roomId:number
}
export {
    Room,
    RoomBooking,
    RentInfo,
    RentState,
    FilterOption,
    RentStateMap,
    RentOrderInfo,
    RoomBookingOrder,
    RoomBookingState,
    RoomBookingStateMap,
    RoomVisitHistory
}
